resources/functions/metering-service.py

```diff
@@ -0,0 +1,161 @@
+import os
+import http.client
+import json
+import time
+import boto3
+from aws_lambda_powertools import Logger, Tracer
+from aws_lambda_powertools.event_handler import APIGatewayHttpResolver
+from aws_lambda_powertools.event_handler.exceptions import BadRequestError
+from aws_lambda_powertools.event_handler.openapi.params import Path
+from aws_lambda_powertools.logging import correlation_paths
+from aws_lambda_powertools.shared.types import Annotated
+from botocore.exceptions import ClientError
+from http import HTTPStatus
+from urllib.parse import urlencode
+
+# Initialize logger and tracer
+logger = Logger()
+tracer = Tracer()
+
+# Fetch environment variables
+API_KEY_SECRET_NAME = os.environ['API_KEY_SECRET_NAME']
+API_KEY_SECRET_ID = os.environ['API_KEY_SECRET_ID']
+BASE_URL = os.environ['AMBERFLO_BASE_URL']
+
+def fetch_api_key():
+    # Fetch the secret from AWS Secrets Manager
+    try:
+        secrets_client = boto3.client('secretsmanager')
+        response = secrets_client.get_secret_value(SecretId=API_KEY_SECRET_NAME)
+        if 'SecretString' in response:
+            return json.loads(response['SecretString'])[API_KEY_SECRET_ID]
+        else:
+            raise ClientError("SecretString not defined")
+    except KeyError as e:
+        logger.error("API key secret not defined")
+        raise
+    except ClientError as e:
+        logger.error(f'Error retrieving secret: {e}')
+        raise
+
+API_KEY = fetch_api_key()
+
+app = APIGatewayHttpResolver()
+
+@app.post("/meters")
+@tracer.capture_method
+def create_meter():
+    payload = app.current_event.json_body
+    if not all(k in payload for k in ['label', 'meterApiName', 'meterType']):
+        raise BadRequestError("Required properties missing for create meter")
+    return make_api_call('POST', '/meters', payload)
+
+@app.put("/meters/<meterId>")
+@tracer.capture_method
+def update_meter():
+    payload = app.current_event.json_body
+    if not all(k in payload for k in ['label', 'meterApiName', 'meterType']):
+        raise BadRequestError("Required properties missing for update meter")
+    payload.setdefault('id', meterId)
+    return make_api_call('PUT', '/meters', payload)
+
+@app.get("/usage/meterId")
```

Comment:
Should this be /usage/<meterId>?

Reply:
They have defined it like this https://github.com/awslabs/sbt-aws/blob/main/src/control-plane/metering/metering-provider.ts#L76 for now, they will update it to `/usage/<meterId>`.

To keep it generic their rationale was to have a GET endpoint for usage and with a path parameter for specific meter.

Reply:
ok